### PR TITLE
Cleanup: Break down HpackIndexingTable::lookup() into static table lookup & dynamic table lookup

### PR DESCRIPTION
**Pull request by masaori335**
Preparation for following changes of static & dynamic table lookup optimization.

- Break down `HpackIndexingTable::lookup()` into  `HpackStaticTable::lookup()` & `HpackDynamicTable::lookup()`
- General cleanups

### CODE REVIEW

File: proxy/http2/HPACK.cc [hunk 1]
```diff
@@ -181,10 +183,10 @@ static const StaticTable STATIC_TABLE[] = {{"", ""},
...
 static constexpr uint32_t HPACK_HDR_HEAP_THRESHOLD = sizeof(MIMEHdrImpl) + sizeof(MIMEFieldBlockImpl) * (2 + 7 + 15);
 
-/******************
- * Local functions
- ******************/
-static inline bool
+//
+// Local functions
+//
+bool
```
Comment (ywkaras) [inline proxy/http2/HPACK.cc]: Why are you changing this to have global scope?  Did you mean to put it into an anonymous namespace?

**Comment by masaori335** [comment]
Rebased on latest master to fix the conflicts.

**Comment by masaori335** [comment]
[approve ci autest]
[approve ci clang-analyzer]

**Comment by masaori335** [comment]
@ywkaras Since you got the power, could you approve this again? I just rebased on the latest master.

**Pull request merged**